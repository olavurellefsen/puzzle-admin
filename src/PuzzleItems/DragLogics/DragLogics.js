import React from "react";
import { DragLogicsContainer, DragLogicsImage } from "./DragLogics.style";

export default () => (
  <DragLogicsContainer data-testid="DragLogicsContainer">
    <DragLogicsImage
      src={"https://via.placeholder.com/200x800?text=Logic"}
      data-testid="DragLogicsImage"
    />
  </DragLogicsContainer>
);
