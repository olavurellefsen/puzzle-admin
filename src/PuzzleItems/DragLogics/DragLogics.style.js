import styled from "styled-components";

export const DragLogicsContainer = styled.div`
  margin: 0px 10px;
`;

export const DragLogicsImage = styled.img``;
