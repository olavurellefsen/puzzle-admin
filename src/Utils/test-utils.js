import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
