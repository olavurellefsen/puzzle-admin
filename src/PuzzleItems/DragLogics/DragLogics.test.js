import React from "react";
import { render, cleanup } from 'test-utils';
import DragLogics from "./DragLogics";
afterEach(cleanup)

it("should render correctly", () => {
  const { container } = render(<DragLogics />);
  expect(container).toMatchSnapshot();
});

it("should show the DragLogicsContainer element", () => {
  const { queryByTestId } = render(<DragLogics />);
  expect(queryByTestId("DragLogicsContainer")).toBeTruthy();
});

it("should show the DragLogicsImage element", () => {
  const { queryByTestId } = render(<DragLogics />);
  expect(queryByTestId("DragLogicsImage")).toBeTruthy();
});