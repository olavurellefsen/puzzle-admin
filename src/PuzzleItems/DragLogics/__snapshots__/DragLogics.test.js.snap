// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa dUCXaL"
    data-testid="DragLogicsContainer"
  >
    <img
      class="sc-bwzfXH uTxCW"
      data-testid="DragLogicsImage"
      src="https://via.placeholder.com/200x800?text=Logic"
    />
  </div>
</div>
`;
